### DISCUSSION TITLE
Does bank-vault support Openbao?

### DISCUSSION THREAD

**Discussion by zetasoumyadeepp** [bank-vaults/bank-vaults#2594] [Q&A] [answered]
Openbao is a fork of the OSS Hashicorp Vault.
https://thenewstack.io/meet-openbao-an-open-source-fork-of-hashicorp-vault/

Has anyone tried using bank-vault with Openbao?

**Comment by csatib02** [comment, marked as answer]
Hi @zetasoumyadeepp!

We're currently working on integrating openBao into our projects, specifically:

- [Secret-init](https://github.com/bank-vaults/secret-init) This is an enhanced version of [Vault-env](https://github.com/bank-vaults/vault-env) that can dynamically and quickly retrieve secrets from various secret stores besides Vault.

-  [Secrets-webhook](https://github.com/bank-vaults/secrets-webhook) is also an improved version of [Vault-secrets-webhook](https://github.com/bank-vaults/vault-secrets-webhook), which will utilise `Secret-init` instead of `Vault-env`, enhancing its versatility.

- In our CLI-tool, [Bank-vaults](https://github.com/bank-vaults/bank-vaults) we are planning to add openBao aswell.

The main challenge we're facing is that openBao hasn't been released yet, and we're uncertain if there will be any interface changes. If not, we can continue using our [Vault-sdk](https://github.com/bank-vaults/vault-sdk)  for Bao as we currently do.

**Reply by r-it-ae** [reply]
Any new's on this ? I didn't found anything on the doc about this, or proposed workaround in repo issue..

**Reply by csatib02** [reply]
Hey @Ronan-WeScale,

The two new project's are in place, you can try them out, What workaround are you searching for exactly?

**Reply by r-it-ae** [reply]
Ok I will check and try. Asking for workaround like modify the vault image to openbao ;) but I think isn't suffisent

**Reply by adamcirilloufs** [reply]
Are you planning on making an openbao operator as well?

**Reply by csatib02** [reply]
Hey @adamcirilloufs,
I don't believe we are going to implement a Bao operator in the near future. :/

**Reply by cipherboy** [reply]
FWIW, I don't think there should be a need other than naming considerations. OpenBao should continue to be compatible with Vault from a consumption perspective.

**Reply by azalio** [reply]
Hi @csatib02! 
Could you please provide an update on openbao?

**Reply by aidro** [reply]
Hey @csatib02, would you happen to have any kind of update on this matter and a possible implementation of OpenBao for the vault-operator? I've had good experience with your [vault-operator](https://github.com/bank-vaults/vault-operator), however, as Hashicorp' solution is no longer a good fit for us, we seek alternatives. Thanks in advance :)

**Comment by Baboulinet-33** [comment]
Hey, any news about this subject ?

**Comment by csatib02** [comment]
@azalio @aidro @Baboulinet-33

Hey guys, sorry for the very late reply.
Unfortunately there hasn't been much progress here, mainly due to time constraints and a lack of maintenance effort. What I can say is that I'll try to pick up where development left off and start sketching out ideas for how this could be moved forward, on the operator's side, starting with the CLI.

**Reply by aidro** [reply]
Hey @csatib02, thanks for your reply! I've been running the Vault-Operator with OpenBao in our production environment for the past year now and have gathered some experience with how OpenBao integrates with the operator. If it is of any help, I would be more then happy to assist in designing and laying the groundwork for an OpenBao integration. Please let me know if I can be of any assistance :)

**Reply by csatib02** [reply]
Most definitely, please contact me on the CNCF slack: https://cloud-native.slack.com/team/U078RG69S94
